src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp

```diff
@@ -1197,6 +1198,9 @@ bool SlaterDetBuilder::readDetList(xmlNodePtr cur,
         confAttrib.add(tag, "id");
         confAttrib.put(cur);
 
+#ifdef QMC_COMPLEX
+        ValueType ci(ci_real, ci_imag);
```

Comment:
I could not understand why this change makes a difference. Could you explain a bit more?

Reply:
inside the ifdef QMC_COMPLEX, ci_real and ci_imag aren't actually being updated until the confAttrib.put(cur) is called. So where the call was previously, the ci is being initialized as 0+i0 instead of the desired ci_real + i ci_imag. The ValueType ci(ci_real,ci_imag) has to happen after ci_real and ci_imag are actually read in

Reply:
Notice in the code above (around line 1000) when usingCSF is true, the correct behavior is obtained because ValueType ci(ci_real, ci_imag) is called after the put. This only effects the case when usingCSFs is false

Reply:
I see it now. Thanks.